File: Known.Web/src/Frame/Core/sys/SysOrganization.js
Picker.action.organization = {
    title: Language.SelectOrganization, type: 'tree', width: 350, height: 350,
    url: baseUrl + '/System/GetOrganizations'
}

function SysOrganization() {
    var url = {
        DeleteModels: baseUrl + '/System/DeleteOrganizations',
        SaveModel: baseUrl + '/System/SaveOrganization',
        GetOrganizations: baseUrl + '/System/GetOrganizations'
    };

    var tree = new Tree('treeOrganization', {
        url: url.GetOrganizations,
        onClick: function (node) {
            if (node.children) {
                var datas = node.children.map(function (d) { return d.data; });
                view.setGridData(datas);
            }
        }
    });

    var view = new View('SysOrganization', {
        url: url,
        left: tree,
        columns: [
            { field: 'Id', type: 'hidden' },
            { field: 'ParentId', type: 'hidden' },
            { title: Language.Code, field: 'Code', width: '100px', type: 'text', required: true },
            { title: Language.Name, field: 'Name', width: '250px', type: 'text', required: true },
            { title: Language.Note, field: 'Note', type: 'textarea' }
        ],
        refresh: function () {
            tree.reload();
            view.setGridData([]);
        },
        formOption: {
            style: 'form-block', width: 600, height: 300,
            titleInfo: function (d) {
                var item = tree.getNodeData(function (l) { return l.id === d.ParentId; });
                return item ? '- ' + item.Code + '|' + item.Name : '';
            },
            setData: function (e) {
                e.form.Code.setReadonly(e.data.Id !== '');
            }
        },
        gridOption: {
            page: false,
            toolbar: {
                add: function (e) {
                    var node = tree.selectedNode;
                    if (!node) {
                        Layer.tips(Language.PleaseSelectParentOrg);
                        return;
                    }
                    e.addRow({ Id: '', ParentId: node.id });
                }
            }
        }
    });

    //methods
    this.render = function (dom) {
        view.render().appendTo(dom);
    }

    this.mounted = function () {
        view.load();
    }
}

$.extend(Page, {
    SysOrganization: { component: new SysOrganization() }
});

File: Known.Core/Entities/SysOrganization.cs
namespace Known.Entities;

/// <summary>
/// 组织架构实体类。
/// </summary>
[DisplayName("组织架构")]
public class SysOrganization : EntityBase
{
    /// <summary>
    /// 取得或设置上级组织。
    /// </summary>
    [MaxLength(50)]
    [DisplayName("上级组织")]
    public string ParentId { get; set; }

    /// <summary>
    /// 取得或设置代码。
    /// </summary>
    [Required]
    [MaxLength(50)]
    [DisplayName("代码")]
    public string Code { get; set; }

    /// <summary>
    /// 取得或设置名称。
    /// </summary>
    [Required]
    [MaxLength(50)]
    [DisplayName("名称")]
    public string Name { get; set; }

    /// <summary>
    /// 取得或设置管理者。
    /// </summary>
    [MaxLength(50)]
    [DisplayName("管理者ID")]
    public string ManagerId { get; set; }

    /// <summary>
    /// 取得或设置备注。
    /// </summary>
    [Column]
    [DisplayName("备注")]
    public string Note { get; set; }
}